src/pathfinder/water_regions.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ /** @file water_regions.h Handles dividing the water in the map into regions to assist pathfinding. */
+
+#ifndef WATER_REGIONS_H
+#define WATER_REGIONS_H
+
+#include "tile_type.h"
+#include "map_func.h"
+
+using TWaterRegionPatchLabel = uint8_t;
+using TWaterRegionIndex = uint;
+
+constexpr int WATER_REGION_EDGE_LENGTH = 16;
+constexpr int WATER_REGION_NUMBER_OF_TILES = WATER_REGION_EDGE_LENGTH * WATER_REGION_EDGE_LENGTH;
+
+/**
+ * Describes a single interconnected patch of water within a particular water region.
+ */
+struct WaterRegionPatchDesc
+{
+	int x; ///< Water region X coordinate
+	int y; ///< Water region Y coordinate
```

Comment:
This and `WaterRegionDesc` both call these `Water region ? coordinate`. However, as far as I could figure out they are not the same. I think for `WaterRegionPatchDesc` they are essentially `Tile X/Y % WATER_REGION_EDGE_LENGTH`, and for `WaterRegionDesc` they are essentially `Tile X/Y / WATER_REGION_EDGE_LENGTH`. 
To avoid confusion, I would give them a different name, and maybe write in the comment of `WaterRegionPatchDesc` something along the line of `Coordinates are within region, i.e. x=2 is the third tile within the region along the X-axis`.
For `WaterRegionDesc` maybe something along the line of `Coordinates are of region, i.e. x=2 is the third region along the X-axis.`

Reply:
They are in fact the same, both are `TileX/Y / WATER_REGION_EDGE_LENGTH`... but I understand the confusion. I've changed the comments to better reflect their meaning based on your suggestion.